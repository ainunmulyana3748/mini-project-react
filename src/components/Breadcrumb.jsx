// src/components/Breadcrumb.jsx
import React from "react";
import { Link } from "react-router-dom";
import { ChevronRight } from "lucide-react";

const Breadcrumb = ({ items }) => {
  return (
    <nav
      className="flex items-center justify-center text-sm text-gray-600 mb-4"
      aria-label="Breadcrumb"
    >
      {items.map((item, index) => (
        <div key={index} className="flex items-center">
          {index > 0 && <ChevronRight className="mx-2 h-4 w-4 text-gray-400" />}
          {item.href ? (
            <Link
              to={item.href}
              className="hover:underline hover:text-black font-medium"
            >
              {item.label}
            </Link>
          ) : (
            <span className="text-gray-500 font-medium">{item.label}</span>
          )}
        </div>
      ))}
    </nav>
  );
};

export default Breadcrumb;
