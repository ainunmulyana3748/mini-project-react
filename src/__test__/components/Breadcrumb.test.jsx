import { render, screen } from "@testing-library/react";
import { BrowserRouter } from "react-router-dom";
import Breadcrumb from "../../components/Breadcrumb";

describe("Breadcrumb Component", () => {
  const items = [
    { label: "Home", href: "/" },
    { label: "Dashboard", href: "/dashboard" },
    { label: "Current Page" },
  ];

  test("render semua item breadcrumb dengan benar", () => {
    render(
      <BrowserRouter>
        <Breadcrumb items={items} />
      </BrowserRouter>
    );
    expect(screen.getByText("Home")).toBeInTheDocument();
    expect(screen.getByText("Dashboard")).toBeInTheDocument();
    expect(screen.getByText("Current Page")).toBeInTheDocument();
  });

  test("item dengan href menjadi link, tanpa href menjadi span", () => {
    render(
      <BrowserRouter>
        <Breadcrumb items={items} />
      </BrowserRouter>
    );
    expect(screen.getByText("Home").closest("a")).toHaveAttribute("href", "/");
    expect(screen.getByText("Dashboard").closest("a")).toHaveAttribute(
      "href",
      "/dashboard"
    );
    expect(screen.getByText("Current Page").tagName).toBe("SPAN");
  });

  test("snapshot breadcrumb", () => {
    const { container } = render(
      <BrowserRouter>
        <Breadcrumb items={items} />
      </BrowserRouter>
    );
    expect(container).toMatchSnapshot();
  });
});
