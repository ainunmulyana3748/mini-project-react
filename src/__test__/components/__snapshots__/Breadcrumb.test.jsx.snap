// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Breadcrumb Component snapshot breadcrumb 1`] = `
<div>
  <nav
    aria-label="Breadcrumb"
    class="flex items-center justify-center text-sm text-gray-600 mb-4"
  >
    <div
      class="flex items-center"
    >
      <a
        class="hover:underline hover:text-black font-medium"
        data-discover="true"
        href="/"
      >
        Home
      </a>
    </div>
    <div
      class="flex items-center"
    >
      <svg
        aria-hidden="true"
        class="lucide lucide-chevron-right mx-2 h-4 w-4 text-gray-400"
        fill="none"
        height="24"
        stroke="currentColor"
        stroke-linecap="round"
        stroke-linejoin="round"
        stroke-width="2"
        viewBox="0 0 24 24"
        width="24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="m9 18 6-6-6-6"
        />
      </svg>
      <a
        class="hover:underline hover:text-black font-medium"
        data-discover="true"
        href="/dashboard"
      >
        Dashboard
      </a>
    </div>
    <div
      class="flex items-center"
    >
      <svg
        aria-hidden="true"
        class="lucide lucide-chevron-right mx-2 h-4 w-4 text-gray-400"
        fill="none"
        height="24"
        stroke="currentColor"
        stroke-linecap="round"
        stroke-linejoin="round"
        stroke-width="2"
        viewBox="0 0 24 24"
        width="24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="m9 18 6-6-6-6"
        />
      </svg>
      <span
        class="text-gray-500 font-medium"
      >
        Current Page
      </span>
    </div>
  </nav>
</div>
`;
